### PR TITLE
Add run-unittests-and-lint Job to Various Flows

### PR DESCRIPTION
**Pull request by avidan-H**
## Status
- [x] Ready

## Related Issues
fixes: link to the issue

## Description
Add docker-in-docker service to `.run-unittests-and-lint` hidden job.

## Minimum version of Cortex XSOAR
- [ ] 5.5.0
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: .gitlab/ci/global.yml [hunk 1]
```diff
@@ -116,6 +116,13 @@
...
+  services:
+    - name: docker.artifactory.pan.local/build-tools--image-dind:19.03.12-dind
+      alias: docker
+  variables:
+    DOCKER_HOST: tcp://docker:2375
+    DOCKER_DRIVER: overlay2
+    DOCKER_TLS_CERTDIR: ""
   extends:
     - .default-job-settings
   script:
```
[the remark this answers is not in the document]
Comment (DinaMeylakh) [reply .gitlab/ci/global.yml:128]: something of the sort:
`curl -fsSL https://get.docker.com | sh`

File: .gitlab/ci/global.yml [hunk 2]
```diff
@@ -116,6 +116,13 @@
     paths:
       - /builds/xsoar/content/unit-tests
       - /builds/xsoar/content/artifacts/*
+  services:
+    - name: docker.artifactory.pan.local/build-tools--image-dind:19.03.12-dind
+      alias: docker
+  variables:
+    DOCKER_HOST: tcp://docker:2375
+    DOCKER_DRIVER: overlay2
+    DOCKER_TLS_CERTDIR: ""
```
[the remark this answers is not in the document]
Comment (DinaMeylakh) [reply .gitlab/ci/global.yml:125]: (the last line)

**Review by DinaMeylakh — COMMENTED** [review]
from what I gathered a docker installation is needed to make this work

File: .gitlab/ci/on-push.yml [hunk 3]
```diff
@@ -21,6 +21,22 @@ trigger-private-build:
     - python3 Utils/get_private_build_status.py --github-token $GITHUB_TOKEN
 
 
+run-unittests-and-lint:
+  extends:
+    - .run-unittests-and-lint
+  rules:
+    - if: '$CI_PIPELINE_SOURCE =~ /^(push|contrib)$/'
```
Comment (Itay4) [inline .gitlab/ci/on-push.yml:28]:
```suggestion
```

File: .gitlab/ci/on-push.yml [hunk 4]
```diff
@@ -21,6 +21,22 @@ trigger-private-build:
...
+    - if: '$NIGHTLY'
+
+
+run-validations:
+  extends:
+    - .run-validations
+  rules:
+    - if: '$CI_PIPELINE_SOURCE =~ /^(push|contrib)$/'
```
Comment (Itay4) [inline .gitlab/ci/on-push.yml:36]:
```suggestion
```

**Review by Itay4 — APPROVED** [review]

**Comment by avidan-H** [comment]
Adding the `docs-approved` label since there is no documentation requiring review from a technical writer.

**Pull request merged**